pkg/daemon/writer.go

```diff
@@ -200,3 +223,48 @@ func (w *NodeStateStatusWriter) writeCheckpointFile(ns *sriovnetworkv1.SriovNetw
 	}
 	return nil
 }
+
+func (w *NodeStateStatusWriter) getCheckPointNodeState(destDir string) (*sriovnetworkv1.SriovNetworkNodeState, error) {
+	glog.Infof("getCheckPointNodeState()")
+	configdir := filepath.Join(destDir, CheckpointFileName)
+	file, err := os.OpenFile(configdir, os.O_RDONLY, 0644)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	defer file.Close()
+	if err = json.NewDecoder(file).Decode(&utils.InitialState); err != nil {
+		return nil, err
+	}
+
+	return &utils.InitialState, nil
+}
+
+func (w *NodeStateStatusWriter) createCleanUpServiceIfNeeded() error {
```

Comment:
Do you want this to work on RHCOS nodes? I think we need to use MCO to inject the systemd service for RHCOS nodes.

Reply:
Hi @pliurh I test this one also on RHCOS and it's working the service is created.

another option will be for the sriov operator to create this system service always for masters and workers pools is that a valid solution?